packages/editor/src/components/provider/index.js

```diff
@@ -111,45 +147,79 @@ function useBlockEditorProps( post, template, mode ) {
 		useEntityBlockEditor( 'postType', template?.type, {
 			id: template?.id,
 		} );
-	const blocks = useMemo( () => {
-		if ( post.type === 'wp_navigation' ) {
-			return [
-				createBlock( 'core/navigation', {
-					ref: post.id,
-					// As the parent editor is locked with `templateLock`, the template locking
-					// must be explicitly "unset" on the block itself to allow the user to modify
-					// the block's content.
-					templateLock: false,
-				} ),
-			];
+	const postContentBlock =
+		extractPostContentBlockFromTemplateBlocks( templateBlocks );
+	const postContentLayout = usePostEditorLayout( postContentBlock );
+
+	const maybeNavigationBlock = useMemo( () => {
+		if ( post.type !== 'wp_navigation' ) {
+			return undefined;
 		}
+		return [
+			createBlock( 'core/navigation', {
+				ref: post.id,
+				// As the parent editor is locked with `templateLock`, the template locking
+				// must be explicitly "unset" on the block itself to allow the user to modify
+				// the block's content.
+				templateLock: false,
+			} ),
+		];
+	}, [ post.type, post.id ] );
 
-		if ( mode === 'post-only' ) {
-			const postContentBlocks =
-				extractPageContentBlockTypesFromTemplateBlocks(
-					templateBlocks
-				);
-			return [
-				createBlock(
-					'core/group',
-					{
-						layout: { type: 'constrained' },
-						style: {
-							spacing: {
-								margin: {
-									top: '4em', // Mimics the post editor.
-								},
+	const maybePostOnlyModeBlocks = useMemo( () => {
+		if ( mode !== 'post-only' ) {
+			return undefined;
+		}
+		const postContentBlocks =
+			extractPageContentBlockTypesFromTemplateBlocks( templateBlocks );
+		const innerBlocks = postContentBlocks.length
+			? postContentBlocks
+			: [
+					createBlock( 'core/post-title' ),
+					createBlock( 'core/post-content' ),
+			  ];
+		const innerBlocksWithLayout = innerBlocks.map( ( block ) => {
+			if ( block.name === 'core/post-content' ) {
+				return createBlock( 'core/post-content', {
+					layout: postContentLayout,
+				} );
+			}
+			return createBlock(
+				'core/group',
+				{
+					layout: postContentLayout,
+				},
+				[ block ]
+			);
```

Comment:
Oh, nice, so this wraps any other content blocks in a Group block that uses the same layout as the post content block 👍

I suppose one thing we might want to look into separately (not a blocker for this PR) could be to also attempt to grab the styles or wrapper for the post title block in case a template is centering the post title or doing other styling to it? For the moment, I think going with making the layout consistent with the post content layout is a good way to go.